src/averageAges.js

```diff
@@ -55,7 +77,35 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const checkMother = motherName => people.find(
+    person => person.name === motherName
+  );
+  const manChecking = person => person.sex === 'm';
+
+  const children = onlyWithSon === true
+    ? people.filter(person => checkMother(person.mother) && manChecking(person))
+    : people.filter(person => checkMother(person.mother));
+
+  const ageDiff = children.map(
+    child => child.born - findMother(child.mother).born
+  );
+
+  function findMother(motherName) {
```

Comment:
Remove the `findMother` function since it is a duplicate of the `checkMother` function.